lib/langchain/llm/openai.rb

```diff
@@ -116,19 +116,20 @@ def chat(prompt: "", messages: [], context: "", examples: [], **options)
       parameters = compose_parameters @defaults[:chat_completion_model_name], options
```

Comment:
What if we actually exposed another method, like `chat_with_functions` just for OpenAI?

Reply:
Not sure about this, as this is an actual option of the chat prompt. why would you separate it on a specific function? 
My reasoning is that probably Open Ai will add more options and this library should be able to scale with that, without the need for more specific options. Also, I like the unified approach, adding another function for one API will kinda diverge the main idea.